pandas/core/arrays/period.py

```diff
@@ -71,6 +87,10 @@ def wrapper(self, other):
         elif other is NaT:
             result = np.empty(len(self._ndarray_values), dtype=bool)
             result.fill(nat_result)
+        elif isinstance(other, (list, np.ndarray)):
```

Comment:
This should be doable I think, assuming that the array can be converted to a PeriodArray.